benchmarks/benchmarks/optimize_linprog.py

```diff
@@ -27,7 +49,40 @@ def klee_minty(D):
     A_ub = toeplitz(A_1, A1_)
     b_ub = np.array([5**(i + 1) for i in range(D)])
     c = -np.array([2**(D - i - 1) for i in range(D)])
-    return c, A_ub, b_ub
+    xf = np.zeros(D)
+    xf[-1] = 5**D
+    obj = c @ xf
+    return c, A_ub, b_ub, xf, obj
+
+
+class MagicSquare(Benchmark):
+
+    params = [
+        methods,
+        [(3, 1.7305505947214375), (4, 1.5485271031586025),
+         (5, 1.807494583582637), (6, 1.747266446858304)]
+    ]
+    param_names = ['method', '(dimensions, objective)']
+
+    def setup(self, meth, prob):
+        dims, obj = prob
+        self.A_eq, self.b_eq, self.c, numbers = magic_square(dims)
+        self.fun = None
+
+    def time_magic_square(self, meth, prob):
+        method, options = meth
+        with suppress_warnings() as sup:
+            sup.filter(OptimizeWarning, "A_eq does not appear")
+            res = linprog(c=self.c, A_eq=self.A_eq, b_eq=self.b_eq,
+                          bounds=(0, 1), method=method, options=options)
+            self.fun = res.fun
+
+    def track_magic_square(self, meth, prob):
+        dims, obj = prob
+        if not self.fun:
+            self.time_magic_square(meth, prob)
+        self.error = np.abs(self.fun - obj)
+        return self.error
```

Comment:
Does this look like a reasonable way of tracking the error in the objective function? Are `time_` benchmarks guaranteed to occur before `track_` benchmarks? (Anecdotally, it seems so.)